### PR TITLE
Adapt latest PF settings in Phase2 Egamma HLT

### PR DESCRIPTION
**Pull request by swagata87**
#### PR description:
This PR is an attempt to align the parameters in 
 - `hltParticleFlowClusterHCALForEgamma` and `particleFlowClusterHCAL`
 - `hltParticleFlowClusterHBHEForEgamma` and `particleFlowClusterHBHE`
 - `hltParticleFlowRecHitHBHEForEgamma` and `particleFlowRecHitHBHE`
so that one of them can be deleted.

With this PR, Egamma will be using 
1) `PFRecHitHCALDenseIdNavigator` (instead of `PFRecHitHCALNavigator`), 
2) `PFRecHitQTestHCALThresholdVsDepth` (instead of `PFRecHitQTestThreshold`),
3) `logWeightDenominatorByDetector` (instead of only `logWeightDenominator`)

(2) is expected to increase timing, so as a safeguard `maxIterations` is decreased from 50 to 5, similar to what was done for Run3 https://its.cern.ch/jira/browse/CMSHLT-2455

Also, this PR addresses https://github.com/cms-sw/cmssw/pull/40525#issuecomment-1383694806

Some changes in trigger results are expected.

#### PR validation:
For 1000 TTbar NoPU events, only change in result is this:
`HLT_PFPuppiMETTypeOne140_PFPuppiMHT140`  NPassed  `25->24`

[With thanks to @hatakeyamak for useful feedback]

### CODE REVIEW

File: HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
         ),
         clusterTimeResFromSeed = cms.bool(False),
         excludeOtherSeeds = cms.bool(True),
-        maxIterations = cms.uint32(50),
+        maxIterations = cms.uint32(5),
```
Comment (missirol) [inline HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py:62]: As far as I understand, this changes the physics outputs of the Jet/MET HLT Phase-2 reco. Should I assume this change is okay with Jet/MET ?

Comment (swagata87) [reply HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py:62]: I think it would be good if they explicitly sign it off.

**Comment by swagata87** [comment]
this is now rebased on top of `CMSSW_13_1_X_2023-03-14-2300` which includes other Phase2 HLT config cleanups: https://github.com/cms-sw/cmssw/pull/40912 and https://github.com/cms-sw/cmssw/pull/41047

File: HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py (hunk 1 above)
Comment (missirol) [reply HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py:62]: @alpakpinar @theochatzis , I guess we wait for your feedback.

Comment (theochatzis) [reply HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py:62]: Hello, we did some preliminary tests on performance and some thresholds for changing the iterations (you can see the results on the slides [here](https://indico.cern.ch/event/1265496/contributions/5314462/subcontributions/418047/attachments/2613316/4515832/preliminaryChecks.pdf) ). From these we see no big effect from this change.

Comment (beaucero) [reply HLTrigger/Configuration/python/HLT_75e33/modules/hltParticleFlowClusterHBHE_cfi.py:62]: Thanks a lot Theo! Please @Martin-Grunewald and @missirol consider that you have the green light from HLT Upgrade!

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
+hlt

Validated by POGs (see PR description, and https://github.com/cms-sw/cmssw/pull/41051#discussion_r1139170663). No changes seen in the outputs of PR tests, likely because HLT-DQM in Phase-2 wfs is largely missing, see #39362.

**Comment by perrotta** [comment]
+1

**Pull request merged**